lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -475,6 +493,47 @@ ExhaustiveSynthesis::synthesize(SMTLIBSolver *SMTSolver,
     llvm::errs() << "There are " << Guesses.size() << " Guesses\n";
 
 
+  std::map<Inst *, Inst *> InstCache;
+  std::map<Block *, Block *> BlockCache;
+
+  if (UseAlive) {
+    Inst *Ante = IC.getConst(APInt(1, true));
+    for (auto PC : PCs ) {
+      Inst *Eq = IC.getInst(Inst::Eq, 1, {PC.LHS, PC.RHS});
+      Ante = IC.getInst(Inst::And, 1, {Ante, Eq});
+    }
+
+    AliveDriver Verifier(LHS, Ante);
+    for (auto &&G : Guesses) {
+      auto C = findConst(G, "reservedconst_0");
+      if (!C) {
+        auto Result = Verifier.verify(G);
+        if (Result) {
+          RHS = G;
+          return EC;
+        }
+      } else {
+        auto Result = Verifier.synthesizeConstant(G);
+        // TODO(manasij7479): Extend to multiple constants
+        if (Result.has_value()) {
+          if (C) {
+            std::map<Inst *, llvm::APInt> ConstMap =
+              {{C, llvm::APInt(C->Width, Result.value())}};
+            auto GWithC = getInstCopy(G, IC, InstCache, BlockCache, &ConstMap, false);
+            auto ResultC = Verifier.verify(GWithC);
+            if (ResultC) {
```

Comment:
there are a few places like here where you create a temp variable and use it in an if-condition, and then it isn't used anywhere else. this would be clearer if you just put the verify call inside the if-condition.